src/guess-game.js

```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { useState } from "react";
+
+export default function GuessGame(props) {
+    const { cheatProbability } = props;
+    const [guess, setGuess] = useState(0);
+    const targetNumber = Math.floor(Math.random() * 1000) + 1;
+    const [target] = useState(targetNumber);
+    const [cheat] = useState(Math.random);
+
+    const onInputChange = (e) => setGuess(Number(e.target.value));
+
```

Comment:
Right now, the reason the message changes is because the `guess` state changes and makes a re-render.
But this is not exactly what we want, and we don't even need a controlled input.

A better way to structure this would be to use a state instead of the `message()` function.
This way, you can control when the message is actually changed and it is not a side effect.
Like this:

```js
const [message, setMessage] = useState("Guess a number");
const targetNumber = Math.floor(Math.random() * 1000) + 1;

const onInputChange = (e) => {
  const guess = e.target.value
  const cheat = Math.random() < cheatProbability;
  const msg = (guess === targetNumber) ? "bingo!" :
        (guess > targetNumber && !cheat) || (guess < targetNumber && cheat) ?
        "too high" : "too low";

  setMessage(msg)
};


return (
  <label><input type="Number" onChange={onInputChange} />{message}</label>
);
```